app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -82,36 +92,48 @@ class EISConnectorImpl(
         Future.failed(new IllegalStateException(s"An unexpected error occurred - got a ${x.getClass}"))
     }
 
-  override def post(body: Source[ByteString, _], hc: HeaderCarrier): Future[Either[RoutingError, Unit]] =
+  private def extractHeaders(movementId: MovementId, messageId: MessageId, hc: HeaderCarrier): Seq[(String, String)] = {
+    val dateHeader: Seq[(String, String)] = hc.headers(Seq(HeaderNames.DATE)) match {
+      case Seq() => Seq(HeaderNames.DATE -> HTTP_DATE_FORMATTER.format(OffsetDateTime.now()))
```

Comment:
Same outcome, just wonder if this is more readable/concise? 

`      val dateHeader = Seq((HeaderNames.DATE, request.headers.get(HeaderNames.DATE).getOrElse(HTTP_DATE_FORMATTER.format(OffsetDateTime.now()))))
`

Reply:
In fact, you could just add it to the headers:
`val date = request.headers.get(HeaderNames.DATE).getOrElse(HTTP_DATE_FORMATTER.format(OffsetDateTime.now()))))`

`HeaderNames.DATE -> date`

Reply:
The issue is that we end up with duplicated headers if we just add it -- something I was trying to avoid (it was happening before this refactor). 

`getOrElse` won't work -- it returns an empty list, not an optional.

Reply:
Ah! Gotcha, thanks :)